woocommerce/woocommerce-ios
# Order Editing: Add save action on Edit Address form + refactor changes detection

Part of https://github.com/woocommerce/woocommerce-ios/issues/4778, https://github.com/woocommerce/woocommerce-ios/issues/4783

## Description

This PR adds save action on edit address form with simulated progress.

### Technical Details

Network action and navbar item logic are straightforward and taken mostly from `EditCustomerNote`.
Complex part was refactoring to have all fields connected through Combine to `updatedAddress` and then to `navigationTrailingItem`. Apart from ugly `CombineLatest` syntax final solution worked well in my tests.

## Screenshots

<img width=350 src="https://user-images.githubusercontent.com/3132438/132768993-706fa164-7bc8-42fb-b99e-9b2f6d06d60d.png">

## Testing

1. Open existing order details and tap edit button for shipping address
2. Observe correct address data in form (except State and Country)
3. Try updating any fields
4. Check that "Done" button in navbar is enabled if there are any unsaved changes
5. Tap "Done" button and observe loading indicator (simulated 3s action)
6. Revert fields to original data
7. Check that "Done" button is disabled

---
Update release notes:
